blenderproc/python/loader/BopLoader.py

```diff
@@ -329,122 +321,38 @@ def get_loaded_obj(model_path: str) -> Optional[bpy.types.Object]:
 
         :param obj_id: The obj_id of the BOP Object.
         :param model_p: model parameters defined in dataset_params.py in bop_toolkit.
         :param bop_dataset_name: The name of the used bop dataset.
-        :param has_external_texture: Set to True, if the object has an external texture.
-        :param temp_dir: A temp directory which is used for writing the temporary .ply file.
         :param allow_duplication: If True, the object is duplicated if it already exists.
         :param scale: factor to transform set pose in mm or meters.
         :return: Loaded mesh object.
         """
 
-        model_path = model_p['model_tpath'].format(**{'obj_id': obj_id})
-
-        texture_file_path = ""  # only needed for ycbv objects
+        model_path = model_p["model_tpath"].format(**{"obj_id": obj_id})
 
         # Gets the objects if it is already loaded
         cur_obj = _BopLoader.get_loaded_obj(model_path)
         # if the object was not previously loaded - load it, if duplication is allowed - duplicate it
         if cur_obj is None:
-            if has_external_texture:
-                if os.path.exists(model_path):
-                    new_file_ply_content = ""
-                    with open(model_path, "r", encoding="utf-8") as file:
-                        new_file_ply_content = file.read()
-                        texture_pos = new_file_ply_content.find("comment TextureFile ") + len("comment TextureFile ")
-                        texture_file_name = new_file_ply_content[texture_pos:
-                                                                 new_file_ply_content.find("\n", texture_pos)]
-                        texture_file_path = os.path.join(os.path.dirname(model_path), texture_file_name)
-                        new_file_ply_content = new_file_ply_content.replace("property float texture_u",
-                                                                            "property float s")
-                        new_file_ply_content = new_file_ply_content.replace("property float texture_v",
-                                                                            "property float t")
-                    model_name = os.path.basename(model_path)
-                    tmp_ply_file = os.path.join(temp_dir, model_name)
-                    with open(tmp_ply_file, "w", encoding="utf-8") as file:
-                        file.write(new_file_ply_content)
-                    bpy.ops.import_mesh.ply(filepath=tmp_ply_file)
-                    cur_obj = bpy.context.selected_objects[-1]
-            else:
-                bpy.ops.import_mesh.ply(filepath=model_path)
-                cur_obj = bpy.context.selected_objects[-1]
-        elif allow_duplication:
+            objs = load_obj(model_path)
+            assert (
+                len(objs) == 1
+            ), f"Loading object from '{model_path}' returned more then one mesh"
+            cur_obj = objs[0]
+        elif False:
```

Comment:
This can probably be removed

Reply:
This was added for debugging, @MartinSmeyer and I tried to find out why some of the materials were loaded differently for objects that appear more then once in a scene. We are still trying to debug this.